Automattic/wp-calypso
# NUX: Add site preview flow and step config

## What this PR does

Here we're integrating the site preview into the `onboarding-dev` flow (now _featuring the 'site style' step!_).

Borrowing from the work in #29400, this PR shifts the code that determines when to show the preview steps, and therefore the site mock up component, to the config.

![jan-02-2019 17-21-28](https://user-images.githubusercontent.com/6458278/50582026-4c16ac80-0eb3-11e9-949f-30f119946467.gif)

## Testing
1. Head to http://calypso.localhost:3000/start/onboarding-dev
2. Navigate through the flow. Ensure that the site preview flow/style steps are displayed correctly.

This work does not effect the `onboarding` flow
